Source/SwiftLintFramework/Rules/Lint/BlanketDisableCommandRule.swift

```diff
@@ -0,0 +1,198 @@
+        kind: .lint,
+        nonTriggeringExamples: [
+            Example("""
+            // swiftlint:disable unused_import
+            // swiftlint:enable unused_import
+            """),
+            Example("""
+            // swiftlint:disable unused_import unused_declaration
+            // swiftlint:enable unused_import
+            // swiftlint:enable unused_declaration
+            """),
+            Example("// swiftlint:disable:this unused_import"),
+            Example("// swiftlint:disable:next unused_import"),
+            Example("// swiftlint:disable:previous unused_import")
+        ],
+        triggeringExamples: [
+            Example("// swiftlint:disable ↓unused_import"),
+            Example("""
+            // swiftlint:disable unused_import ↓unused_declaration
+            // swiftlint:enable unused_import
+            """),
+            Example("""
+            // swiftlint:disable unused_import
+            // swiftlint:disable ↓unused_import
+            // swiftlint:enable unused_import
+            """),
+            Example("""
+            // swiftlint:enable ↓unused_import
+            """)
+        ].skipWrappingInCommentTests().skipDisableCommandTests()
+     )
+
+    func validate(file: SwiftLintFile) -> [StyleViolation] {
+        var violations: [StyleViolation] = []
+        var ruleIdentifierToCommandMap: [RuleIdentifier: Command] = [:]
+        var disabledRuleIdentifiers: Set<RuleIdentifier> = []
+
+        for command in file.commands {
+            if command.action == .disable {
+                violations += validateAlreadyDisabledRules(
+                    for: command,
+                    in: file,
+                    disabledRuleIdentifiers: disabledRuleIdentifiers
+                )
+            }
+
+            if command.action == .enable {
+                violations += validateAlreadyEnabledRules(
+                    for: command,
+                    in: file,
+                    disabledRuleIdentifiers: disabledRuleIdentifiers
+                )
+            }
+
+            if command.modifier != nil {
+                continue
+            }
+
+            if command.action == .disable {
+                disabledRuleIdentifiers.formUnion(command.ruleIdentifiers)
+                command.ruleIdentifiers.forEach { ruleIdentifierToCommandMap[$0] = command }
+            }
+            if command.action == .enable {
+                disabledRuleIdentifiers.subtract(command.ruleIdentifiers)
+                command.ruleIdentifiers.forEach { ruleIdentifierToCommandMap.removeValue(forKey: $0) }
+            }
+        }
+
+        violations += validateBlanketDisables(
+            in: file,
+            disabledRuleIdentifiers: disabledRuleIdentifiers,
+            ruleIdentifierToCommandMap: ruleIdentifierToCommandMap
+        )
+        violations += validateAlwaysBlanketDisable(file: file)
+
+        return violations
+    }
+
+    private func violation(
```

Comment:
Is this indirection really needed for only one saved conversion?

Reply:
So in three cases I have a RuleIdentifier (so they use the indirection), and then in two cases, in `validateAlwaysBlanketDisable`, I have the string.

I couldn't see anyway round that - the indirection felt cleaner than making the three call sites call `stringRepresentation` themselves.